tests/cocotb/tests/test_rv32i_debug.py

```
@@ -0,0 +1,269 @@
+    env.load_program(hex_file)
+
+    # Set a breakpoint at a specific address
+    bp_addr = 0x0010  # Set breakpoint at PC=0x10
+    env.log.info(f"Setting breakpoint 0 at address 0x{bp_addr:08x}")
+    await env.apb_agent.set_breakpoint(0, bp_addr, enable=True)
+
+    # Let the CPU run
+    env.log.info("Running CPU until breakpoint")
+
+    # Wait for CPU to halt at breakpoint (with timeout)
+    for cycle in range(1000):
```

Ruff (0.14.11)

Loop control variable `cycle` not used within loop body

Rename unused `cycle` to `_cycle`

(B007)